### PR TITLE
Document upcoming breaking changes

### PR DESCRIPTION
**Pull request by treydock**
# Pull Request Checklist

## Description

This is a way to document all the breaking changes that will occur with 4.x of this module. This will hopefully give a very clear way for people identify changes they must make to support 4.x of this module.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,6 +91,33 @@ sensu_asset { 'test':
 }
 ```
```
Comment (ghoneycutt) [inline README.md]: should update the `Compatibility` section with a pointer to sensu classic as well as sensu go 5.16+ to use v4

Comment (treydock) [reply README.md]: Change added

File: README.md [hunk 2]
```diff
@@ -46,15 +47,19 @@ Please note, that this is a **Partner Supported** module, which means that techn
...
+| 5.16+           | latest v4                           |
 
 ### Upgrade note
 
-Sensu Go 5.x is a rewrite of Sensu and no longer depends on redis and rabbitmq. Version 3 of this module supports Sensu Go 5.x.
+Sensu Go 5.x is a rewrite of Sensu and no longer depends on redis and rabbitmq.
+Version 3 of this module supports Sensu Go 5.0.0 to 5.15.x.
+Version 4 of this module supports Sensu Go 5.16+.
 
 Users wishing to use the old v2 Puppet module to support previous Ruby based Sensu should use [sensu/sensuclassic](https://forge.puppet.com/sensu/sensuclassic).
```
Comment (ghoneycutt) [inline README.md]: Spotted this sentence that reads funny. Could you update it with these other changes.

```
Users wishing to use the previous Ruby based Sensu should use the [sensu/sensuclassic](https://forge.puppet.com/sensu/sensuclassic) module.
```

Comment (treydock) [reply README.md]: Change made

File: README.md [hunk 3]
```diff
@@ -46,17 +47,21 @@ Please note, that this is a **Partner Supported** module, which means that techn
 ### Compatibility - supported sensu versions
```
Comment (ghoneycutt) [inline README.md]: Sensu is a proper noun as name of their product and should be capitalized. Similar for Puppet, Redis and RabbitMQ if we are referring to the tool and not a command.

Comment (treydock) [reply README.md]: Fixed

**Pull request merged**